utilities/warning/deprecated-property-value.js

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+/* eslint-disable import/no-mutable-exports */
+/* eslint-disable max-len */
+
+// This function will deliver an error message to the browser console about the future of a removal and moving of a property's valid value to another prop. This makes the most sense to be used with `oneOf` prop types.
+import warning from 'warning';
+
+let deprecated = function () {};
+
+if (process.env.NODE_ENV !== 'production') {
+	const hasWarned = {};
+
+	deprecated = function (
```

Comment:
We should have unit tests for these that verify they work the way we think they work.

Reply:
Good idea. I've created #1337 and will work on a unit test limited to just `deprecated-property-value.js` for this pull request.

Reply:
unit test for only the warning added in this PR created.